client/src/components/Visualization/charts/ScatterPlot.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Description: exports a react/victory object of 
+ *              a scatterplot with the top X most prolific advertisers
+ *              across 2018, 2019, 2020, as computed in `aggregates`
+ *              firestore collection. 
+ * Date: 7/21/2020
+ * Author: Robbie Marcus
+ */
+
+import { database } from '../../../firebase/firebase';
+import React, { useState, useEffect } from 'react';
+import { VictoryTheme, 
+         VictoryScatter, 
+         VictoryChart, 
+         VictoryAxis, 
+         VictoryTooltip, 
+         VictoryLabel } from 'victory';
+
+/**
+ * Creates a hexidecimal color string from a string input. 
+ * Taken from here: `https://stackoverflow.com/a/16348977`.
+ * Loosely cleaned up for style and comments.
+ * This *should* return a valid hash, but if it doesn't,
+ * the browser will just display `#000`.
+ * @param {string} str input string 
+ * @return {string}
+ */
+function stringToColor(str) {
+  let hash = 0;
+  let hexColor = '#';
+
+  // Compute a hash from the input string.
+  for (let index = 0; index < str.length; index++) {
+    hash = str.charCodeAt(index) + ((hash << 5) - hash);
+  }
+  
+  // Compute a hex string from the hash.
+  for (let index = 0; index < 3; index++) {
+    let value = (hash >> (index * 8)) & 0xFF;
+    hexColor += ('00' + value.toString(16)).substr(-2);
+  }
+
+  return hexColor;
+}
+
+/**
+ * Turn a firestore snap into a victory formatted
+ * json object that can be charted without further alteration.
+ * @param {object} snap a firestore snap object
+ * @param {string} year year corresponding to the relevant year. 
+ * @return {object}
+ */
+function formatAdvertiserCountSnapshot(snap, year) {
+  const numberOfAds = snap.data().numberOfAds;
+  const advertiser = snap.id;
+
+  const victoryFormattedAdvertiserCount = 
+      {x: parseInt(year), 
+       y: parseInt(numberOfAds), 
+       fill: stringToColor(advertiser),
+       label: `${advertiser} had ${numberOfAds} ads in ${year}.`};
+
+  return victoryFormattedAdvertiserCount;
+}
+
+/**
+ * Returns the exclusive bounds for the range of all data points being charted.
+ * Uses the spread `...` operator.
+ * `https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax` 
+ * @param {List[object]} victoryJsonList a list of objects in a victory-ready 
+ *     format.
```

Comment:
nit: I think this can be one line

Reply:
80 char limit.

Reply:
Ah ok, the line before had over 80. Different for links?

Reply:
Yup. You don't want to break up links bc then when you try and click on them it won't work. Same for searching for code etc. Adds unnecessary complexity.